esphome/components/ina260/ina260.cpp

```diff
@@ -0,0 +1,143 @@
+#include "ina260.h"
+#include "esphome/core/log.h"
+#include "esphome/core/hal.h"
+
+namespace esphome {
+namespace ina260 {
+
+static const char *const TAG = "ina260";
+
+// | A0   | A1   | Address |
+// | GND  | GND  | 0x40    |
+// | GND  | V_S+ | 0x41    |
+// | GND  | SDA  | 0x42    |
+// | GND  | SCL  | 0x43    |
+// | V_S+ | GND  | 0x44    |
+// | V_S+ | V_S+ | 0x45    |
+// | V_S+ | SDA  | 0x46    |
+// | V_S+ | SCL  | 0x47    |
+// | SDA  | GND  | 0x48    |
+// | SDA  | V_S+ | 0x49    |
+// | SDA  | SDA  | 0x4A    |
+// | SDA  | SCL  | 0x4B    |
+// | SCL  | GND  | 0x4C    |
+// | SCL  | V_S+ | 0x4D    |
+// | SCL  | SDA  | 0x4E    |
+// | SCL  | SCL  | 0x4F    |
+
+enum {
+  INA260_REGISTER_CONFIG = 0x00,
+  INA260_REGISTER_CURRENT = 0x01,
+  INA260_REGISTER_BUS_VOLTAGE = 0x02,
+  INA260_REGISTER_POWER = 0x03,
+  INA260_REGISTER_MASK_ENABLE = 0x06,
+  INA260_REGISTER_ALERT_LIMIT = 0x07,
+  INA260_REGISTER_MANUFACTURE_ID = 0xFE,
+  INA260_REGISTER_DEVICE_ID = 0xFF
+};
+
+void INA260Component::setup() {
+  ESP_LOGCONFIG(TAG, "Setting up INA260...");
+
+  // Reset device on setup
+  // 0bx000000000000000 << 15 RESET Bit (1 -> trigger reset)
+  if (!this->write_byte_16(INA260_REGISTER_CONFIG, 0x8000)) {
+    this->error_code_ = DEVICE_RESET_FAILED;
+    this->mark_failed();
+    return;
+  }
+
+  delay(2);  // NOLINT
```

Comment:
Don't need to unnecessarily bypass lint here.

```suggestion
  delay(2);
```